client/src/components/GameBoard.js

```diff
@@ -32,15 +34,30 @@ const useStyles = makeStyles({
 export default function GameBoard() {
     const classes = useStyles();
     const { email } = useUserState();
-    const { matchState } = useContext(MatchContext);
+    const { matchState, setMatchState } = useContext(MatchContext);
+    const history = useHistory();
+    const socketRef = useRef();
+    const matchId = history.location.pathname.substring(
+        history.location.pathname.lastIndexOf("/") + 1
+    );
```

Comment:
Could you use path parameters to determine this as with the Join page?

Reply:
Done